websites/Z/ZARP Gaming/presence.ts

```diff
@@ -0,0 +1,141 @@
+const presence = new Presence({
+	clientId: "990710812320612432",
+});
+
+presence.on("UpdateData", () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "zarp-512x512",
+	};
+	switch (window.location.host) {
+		case "zarpgaming.com":
+			if (
+				window.location.pathname == "/index.php" ||
+				window.location.pathname == "/index.php/forum"
+			)
+				presenceData.details = "Viewing the home page";
+
+			if (window.location.pathname.startsWith("/index.php/forum/")) {
+				presenceData.details = "Viewing a section:";
+				presenceData.state = (document.querySelector(
+					"#Kunena > form > div > div.kheader > h2 > span"
+				) as HTMLElement)?.innerText || document.title;
+				presenceData.buttons = [
+					{
+						label: "Visit Section",
+						url: window.location.href,
+					},
+				];
+			}
+
+			if (window.location.pathname == "/index.php/servers")
+				presenceData.details = "Viewing servers";
+
+			if (window.location.pathname == "/index.php/user-list")
+				presenceData.details = "Viewing the list of users";
+
+			if (window.location.pathname == "/index.php/zarp-vip")
+				presenceData.details = "Viewing the VIP page";
+
+			if (window.location.pathname == "/index.php/zarp-vip/booster-pack")
+				presenceData.details = "Viewing the booster packs";
+
+			if (window.location.pathname == "/index.php/help")
+				presenceData.details = "Viewing the help page";
+
+			if (window.location.pathname.startsWith("/index.php/help/faq")) {
+				presenceData.details = "Viewing the FAQ";
+				if (window.location.pathname.includes("faq/"))
+					presenceData.state = document.title;
+			}
+
+			if (
+				document.querySelector(
+					"#Kunena > div:nth-child(5) > div.kheader > h1 > span"
+				)
+			) {
+				presenceData.details = "Viewing a thead:";
+				presenceData.state = (document.querySelector(
+					"#Kunena > div:nth-child(5) > div.kheader > h1 > span"
+				) as HTMLElement).innerText.replace("TOPIC: ", "");
```

Comment:
HTMLElement is an abstract class, please use the proper type or none at all. Also don't use `as`